### PR TITLE
Update scalastyle to allow 2021 in the copyright notice

### LINKED ISSUE #382: Scala style does NOT allow 2021 Copyright
We are now in 2021. Scala style check (`sbt clean compile`) should allow this:

```
Copyright (2021) The Hyperspace Project Authors.
...
```

Instead it fails with: 

```
[error] /Users/aionescu/github/hyperspace/src/main/scala/com/microsoft/hyperspace/index/rules/PlanUtils.scala: Header does not match expected text
```

### PR DESCRIPTION
**Pull request by clee704**
### What is the context for this pull request?

 - **Tracking Issue**: #382 
 - **Parent Issue**: N/A
 - **Dependencies**: N/A

Fixes #382

### What changes were proposed in this pull request?

According to my research, it is best to include the creation year of the file in the copyright notice. This change allows writing 2021 in the header.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Locally checked with scalastyle

### CODE REVIEW

**Review by andrei-ionescu — DISMISSED** [review]

File: scalastyle-config.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@ This file is divided into 3 sections:
  */
 
 \E)?\Q/*
- * Copyright (2020) The Hyperspace Project Authors.
+ * Copyright (\E\d{4}(?:-\d{4})?\Q) The Hyperspace Project Authors.
```
Comment (imback82) [inline scalastyle-config.xml:95]: This will not catch invalid years?

Comment (clee704) [reply scalastyle-config.xml:95]: I think the purpose of having a code linter is mainly to catch human errors, and having a strict validation here seems like overkill. But if you want we can make it something like 20\d{2} to accept any 20xx year (although we can't still prevent invalid years like 2099 or 2099-2010), or make it accept only years known to valid for now (2020|2021|2020-2021) and change the format every year.

Comment (imback82) [reply scalastyle-config.xml:95]: I definitely prefer catching this kind of error via scalastyle (why waste time in PR?). We can just do `(2020|2021)` and I don't think it's an overkill to update this every year.

Comment (clee704) [reply scalastyle-config.xml:95]: After some research it seems sufficient to use something like (2020|2021|...|<current year>). When you create a new file, you just write the year at the moment. No need to update old files.

Comment (clee704) [reply scalastyle-config.xml]: By the way, if you really want to be strict, then we should prevent people using past year when creating new files and updating the year of old files - currently we can't catch those errors

Comment (andrei-ionescu) [reply scalastyle-config.xml]: I'm also ok with adding new years as we get into a new year. Also, I would not want to have ranges (ie: `2019-2021`). The year of first publishing should be enough.

**Comment by sezruby** [comment]
LGTM Thanks!

**Review by andrei-ionescu — APPROVED** [review]
LGTM, Thanks!

**Review by imback82 — APPROVED** [review]
LGTM, thanks @clee704!

**Pull request merged**